geometry/optimization/convex_set.h

```diff
@@ -87,20 +89,40 @@ class ConvexSet : public ShapeReifier {
   void AddPointInSetConstraint(
       solvers::MathematicalProgram* prog,
       const Eigen::Ref<const solvers::VectorXDecisionVariable>& vars) const {
+    DRAKE_DEMAND(vars.size() == ambient_dimension());
     return DoAddPointInSetConstraint(prog, vars);
   }
 
+  /** Adds the convex constraints to imply
+  @verbatim
+  x ∈ t S,
```

Comment:
One technical observation, probably my bad, sorry for this. It is not exactly true that this method enforces `x ∈ t S`, because that would imply that `x = 0` if `t = 0`, even if `S` is unbounded.

To make this very precise I see two options:
- Explicitly require the set `S` to be bounded for this method to be used.
- Or say that this enforces the constraint `x ∈ t S + rec(S)` where `rec(S)` is the recession cone of `S` (the asymptotic directions in which `S` is not bounded). In fact, when `t = 0` the perspective is actually enforcing `x ∈ rec(S)` (which is `x ∈ {0}` for a bounded set), while when `t > 0` it enforces `x ∈ t S + rec(S) = t S`.